File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline


def dfs(graph, v, visited):
    visited[v] = True

    for i in graph[v]:
        if not visited[i]:
           dfs(graph, i, visited)
n, m = map(int, input().split())

graph = [[] for _ in range(n + 1)]

for i in range(m):
    x, y = map(int, input().split())
    graph[x].append(y)
    graph[y].append(x)



answer = 0
visited = [False] * (n + 1)

for i in range(1, n + 1):
    if not visited[i]:
        dfs(graph, i, visited)
        answer += 1

print(answer)

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.java

import java.util.*;

public class Main {

	static boolean visited[];
	static List<List<Integer>> list;
	static int answer = 0;
	
	public static void DFS(int v) {
		visited[v] = true;
		
		for(int num : list.get(v)) {
			if(!visited[num]) {
				DFS(num);
			}
		}
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int n = sc.nextInt();
		int m = sc.nextInt();

		visited = new boolean[n + 1];
		list = new ArrayList<>();
		
		for(int i = 0; i <= n; i++) {
			list.add(new ArrayList<>());
		}
		for (int i = 0; i < m; i++) {
			int x = sc.nextInt();
			int y = sc.nextInt();
			list.get(x).add(y);
			list.get(y).add(x);
		}

		for(int i = 1; i < visited.length; i++) {
			if(!visited[i]) {
				DFS(i);
				answer++;
			}
		}
		System.out.println(answer);
	}
}
